includes/payments/functions.php

```diff
@@ -609,6 +609,45 @@ function edd_check_for_existing_payment( $payment_id ) {
 }
 
 /**
+ * Retrieves an instance of EDD_Payment for a specified ID.
+ *
+ * @since 2.7
+ *
+ * @param mixed int|EDD_Payment|WP_Post $payment Payment ID, EDD_Payment object or WP_Post object.
+ * @param bool                          $by_txn  Is the ID supplied as the first parameter
+ * @return mixed false|object EDD_Payment if a valid payment ID, false otherwise.
+ */
+function edd_get_payment( $payment = null, $by_txn = false ) {
+	if ( is_a( $payment, 'WP_Post' ) || is_a( $payment, 'EDD_Payment' ) ) {
+		$payment_id = $payment->ID;
+	}
+
+	if ( is_numeric( $payment ) ) {
+		$payment_id = absint( $payment );
+	} else {
+		$payment_id = $payment;
+	}
+
+	if ( empty( $payment_id ) ) {
+		return false;
+	}
+
+	$cache_key = md5( 'edd_payment' . $payment_id );
+	$_payment = wp_cache_get( $cache_key, 'payments' );
```

Comment:
`$_payment` is never returned when retrieved from cache due to the variable names being different.